File: classes/javascript/Property.js
class ObjectType {
  constructor(value){
    this._propname = value;
  }
  get propname() {
    console.log('getting propname');
    return this._propname;
  }
  set propname(value) {
    console.log('setting propname');
    this._propname = value;
  }
}

const o = new ObjectType('hola');
console.log(o.propname);

o.propname = 'chau';
console.log(o.propname);

File: classes/python/Property.py
#! /usr/bin/env python

class ObjectType:
    def __init__(self, value):
        self._propname = value

    @property
    def propname(self):
        """Propname docstring."""
        print('getting propname')
        return self._propname

    @propname.setter
    def propname(self, value):
        print('setting propname')
        self._propname = value

    @propname.deleter
    def propname(self):
        print('deleting propname')
        del self._propname


o = ObjectType('hola')
print(o.propname)

o.propname = 'chau'
print(o.propname)

del o.propname
print('Has attribute propname?', hasattr(o, 'propname'))
